tests/kafkatest/services/kafka/kafka.py

```diff
@@ -330,7 +327,7 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
             if self.quorum_info.has_brokers:
                 num_nodes_broker_role = num_nodes
                 if self.quorum_info.has_controllers:
-                    self.num_nodes_controller_role = self.num_kraft_controllers(num_nodes_broker_role, controller_num_nodes_override)
+                    self.num_nodes_controller_role = self.num_kraft_controllers(num_nodes_broker_role, 0)
```

Comment:
Pardon me, can assigning the value to `0` address the "Propose to set the controller node to at least 3"? It seems to me, it still depend on the `num_nodes_broker_role`, right?

```
        if num_nodes_broker_role < 3:
            return 1
        if num_nodes_broker_role < 5:
            return 3
        return 5
```

https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/kafka/kafka.py#L495

Reply:
I may be mistaken, but I understand this pull request to be intended to increase the number of controller nodes, thereby preventing a complete controller service outage during a rolling bounce. Is that correct?